src/components/TextInputWithDropdown/TextInputWithDropdown.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { useFormikContext } from 'formik'
+import { CreateProfileFormValues } from '../CreateProfileForm/CreateProfileFormWrapper'
+import TextInput from '../TextInput/TextInput'
+import DropdownCountry from '../Dropdowns/DropdownCountry/DropdownCountry'
+import styles from './TextInputWithDropdown.module.scss'
+import { countries } from '@/data/frontend/profile/countries/countries'
+
+const TextInputWithDropdown = () => {
+  const { values, handleChange, setFieldValue } =
+    useFormikContext<CreateProfileFormValues>()
+  const [isDropdownActive, setIsDropdownActive] = useState(false)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const preventInvalidCountry = () => {
+    const inputValue = inputRef.current?.value
+    if (!isValidCountry(inputValue)) {
+      const validCountry = countries.find((country) =>
+        inputValue?.includes(country.name),
+      )
+      setFieldValue('country', `${validCountry?.name ? validCountry.name : ''}`)
+    }
+  }
+
+  const isValidCountry = (countryName: string | undefined) => {
+    if (countryName) {
+      return countries.some((country) =>
+        country.name.toLowerCase().includes(countryName.toLowerCase()),
+      )
+    }
+  }
+  const handleCountryInputClick = () => {
+    setIsDropdownActive(!isDropdownActive)
+  }
+
+  const handleClickOutside = (e: MouseEvent) => {
+    const dropdownContainer = dropdownRef.current
+    if (dropdownContainer && !dropdownContainer.contains(e.target as Node)) {
+      setIsDropdownActive(false)
+      preventInvalidCountry()
+    }
+  }
+
+  useEffect(() => {
```

Comment:
**nitpick:** We should create some reusable hook called `useOutsideClick` so its possible to use it somewhere else